nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/MoveToColdStorageTest.java

```diff
@@ -0,0 +1,86 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *  
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.ecm.automation.core.operations.document;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.nuxeo.ecm.core.DummyThumbnailFactory.FAKE_THUMBNAIL_CONTENT;
+import static org.nuxeo.ecm.core.api.impl.blob.AbstractBlob.UTF_8;
+import static org.nuxeo.ecm.core.blob.ColdStorageHelper.FILE_CONTENT_PROPERTY;
+import static org.nuxeo.ecm.core.schema.FacetNames.COLD_STORAGE;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.Map;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.core.api.Blob;
+import org.nuxeo.ecm.core.api.Blobs;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@Deploy("org.nuxeo.ecm.automation.core")
+@Deploy("org.nuxeo.ecm.core.test.tests:OSGI-INF/test-coldstorage-contrib.xml")
+public class MoveToColdStorageTest {
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected AutomationService automationService;
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @Test
+    public void shouldMoveToColdStorage() throws OperationException, IOException {
+        DocumentModel documentModel = session.createDocumentModel("/", "MyFile", "File");
+        documentModel.setPropertyValue(FILE_CONTENT_PROPERTY, (Serializable) Blobs.createBlob("foo"));
+        documentModel = session.createDocument(documentModel);
+        assertFalse(documentModel.hasFacet(COLD_STORAGE));
+        try (OperationContext context = new OperationContext(session)) {
+            context.setInput(documentModel);
+            DocumentModel updatedDocModel = (DocumentModel) automationService.run(context, MoveToColdStorage.ID,
+                    Map.of());
+            Blob fileContent = (Blob) documentModel.getPropertyValue(FILE_CONTENT_PROPERTY);
+            txFeature.nextTransaction();
```

Comment:
I think this is not needed since we are in the same transaction (no REST call for instance), in fact, you are retrieving `fileContent` before this call and the assertion that happens after is working...